File: packages/tasks-gen/snippets-fixtures/image-to-image/js/fetch/0.replicate.js
const image = fs.readFileSync("cat.png");

async function query(data) {
	const response = await fetch(
		"https://router.huggingface.co/replicate/v1/models/<replicate alias for black-forest-labs/FLUX.1-Kontext-dev>/predictions",
		{
			headers: {
				Authorization: `Bearer ${process.env.HF_TOKEN}`,
				"Content-Type": "image/jpeg",
			},
			method: "POST",
			body: {
				"inputs": `data:image/png;base64,${data.inputs.encode("base64")}`,
				"parameters": data.parameters,
			}
		}
	);
	const result = await response.json();
	return result;
}

query({ 
	inputs: image,
	parameters: {
		prompt: "Turn the cat into a tiger.",
	}
}).then((response) => {
    console.log(JSON.stringify(response));
});

File: packages/tasks-gen/snippets-fixtures/image-to-image/python/requests/0.replicate.py
import os
import base64
import requests

API_URL = "https://router.huggingface.co/replicate/v1/models/<replicate alias for black-forest-labs/FLUX.1-Kontext-dev>/predictions"
headers = {
    "Authorization": f"Bearer {os.environ['HF_TOKEN']}",
}

def query(payload):
    with open(payload["inputs"], "rb") as f:
        img = f.read()
        payload["inputs"] = base64.b64encode(img).decode("utf-8")
    response = requests.post(API_URL, headers=headers, json=payload)
    return response.content

image_bytes = query({
    "input": {
        "prompt": "Turn the cat into a tiger.",
        "image": "cat.png",
        "images": [
            "cat.png"
        ],
        "input_image": "cat.png",
        "input_images": [
            "cat.png"
        ]
    }
})

# You can access the image with PIL.Image for example
import io
from PIL import Image
image = Image.open(io.BytesIO(image_bytes))